src/RunScriptCommand.cs

```diff
@@ -134,15 +134,15 @@ public async Task<int> InvokeAsync(InvocationContext context)
     }
 
     internal static List<ScriptResult> FindScripts(
-        IDictionary<string, string?> projectScripts,
+        ScriptCollection projectScripts,
         string[] scripts)
     {
         var results = new List<ScriptResult>();
 
         foreach (var script in scripts)
         {
             // The `env` script is special so if it's not explicitly declared we act like it was
-            if (projectScripts.ContainsKey(script) || string.Equals(script, "env", StringComparison.OrdinalIgnoreCase))
+            if (projectScripts.Contains(script) || string.Equals(script, "env", StringComparison.Ordinal))
```

Comment:
If you expose the key comparer of projectScripts, you can use that same comparer to compare script with "env" as a way to ensure that you use the same behavior in both places. If env is special and must always be a special ordinal comparison, I think it should be a dedicated (static) method with a descriptive name and maybe a description regarding why it exists. Similar code is found in CommandGroupRunner too.